### PR TITLE
fix tests: update tests in 04-channel/types , use expected errors

### PR DESCRIPTION
**Pull request by likesToEatFish**
## Description

ref: https://github.com/cosmos/ibc-go/issues/7175

Use expected errors in modules/core/04-channel/types/

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/24-host/parse.go
Comment (gjermundgaraba) [inline modules/core/24-host/parse.go:30]: In these scenario (where there is an actual error), we typically don't wrap like this, we wrap the way this was already doing: `errorsmod.Wrap(err, "failed to parse identifier sequence")`

Comment (likesToEatFish) [reply modules/core/24-host/parse.go:30]: -Consistency: Other errors in the same function are using ErrInvalidID instead of wrapping directly from err. This ensures that the error message is always consistent about the error type ErrInvalidID.

-Error structure: ErrInvalidID is used to represent a specific identifier format error. If wrapped with another external err (errorsmod.Wrap), it may not ensure consistency about the error type.

I suggest the following:
`return 0, errorsmod.Wrapf(ErrInvalidID, "failed to parse sequence: %s", err.Error())`

Are you ok?

Comment (damiannolan) [reply modules/core/24-host/parse.go:30]: For this case either is fine imo, but generally prefer to wrap err returned as @gjermundgaraba suggested

Comment (likesToEatFish) [reply modules/core/24-host/parse.go:30]: Thanks u

> For this case either is fine imo, but generally prefer to wrap err returned as @gjermundgaraba suggested

**Review by DimitrisJim — APPROVED** [review]
Lgtm, will merge after sdk upgrade pr is merged

**Pull request merged**
